### DISCUSSION TITLE
Preserve configurations when connecting via SSH

### DISCUSSION THREAD

**Discussion by Sporesirius** [wezterm/wezterm#3870] [Q&A]
Hello,
Is it possible to keep the configs and styles of e.g. zsh, fish, starship and connect over ssh without sending configs or installing packages on the server side?
Or is it out of scope for wezterm? I saw that wezterm has a built-in ssh client, but when I try to connect to a server the configs, styles are not applied.

Kind regards
Sporesirius

**Comment by bew** [comment]
What config / styles are you talking about?
Where are they?
How are you connecting to the server and what is different?

**Reply by Sporesirius** [reply]
I mean e.g. .config files of zsh, fish, starship. Of course, normally when you connect over ssh, you don't have your local configuration files, so you'll have e.g. on a standard debian remote machine a bash shell with no customizations. I was wondering if there was a way for wezterm to show my local shell with my configs, but interact with the remote machine over ssh. I saw that wezterm has a built-in ssh client and read about a shell integration with wezterm. I may be misunderstanding something here, but this would be very handy.

**Reply by wez** [reply]
when you connect via `ssh` you are asking the remote host to run a remote shell on your behalf.

It sounds like maybe you'd want to consider using something like SSHFS to mount the remote filesystem using SSH/SFTP and then browse that using your local shell.

That is out of scope of wezterm.

**Reply by jankatins** [reply]
Or a dotfile manager: yadm, chezmoi,...

Chezmoi has a comparison: https://www.chezmoi.io/comparison-table/
